openvdb/unittest/TestFastSweeping.cc

```diff
@@ -0,0 +1,635 @@
+            //std::cerr << "Total : " << grid->activeVoxelCount() << std::endl;
+            //CPPUNIT_ASSERT(message.empty());
+            //CPPUNIT_ASSERT_EQUAL(size_t(0), diagnose.failureCount());
+            CPPUNIT_ASSERT(percent < 0.01);
+            //std::cout << "\nOutput 1: " << message << std::endl;
+        }
+    }
+#ifdef TestFastSweeping_DATA_PATH
+     {// Use bunny as a mask
+         //std::cerr << "\nUse bunny as a mask" << std::endl;
+         FloatGrid::Ptr grid = tools::createLevelSetSphere<FloatGrid>(10.0f, Vec3f(-10,0,0), 0.05f, width);
+         openvdb::initialize();//required whenever I/O of OpenVDB files is performed!
+         const std::string path(TestFastSweeping_DATA_PATH);
+         io::File file( path + "bunny.vdb" );
+         file.open(false);//disable delayed loading
+         FloatGrid::Ptr mask = openvdb::gridPtrCast<openvdb::FloatGrid>(file.getGrids()->at(0));
+
+         writeFile("/tmp/bunny_mask_input.vdb", grid);
+         tools::FastSweeping<FloatGrid> fs;
+#ifdef TIMING_FAST_SWEEPING
+         util::CpuTimer timer("\nParallel sparse fast sweeping with a bunny mask");
+#endif
+         fs.initMask(*grid, *mask);
+         //std::cerr << "voxel count = " << fs.voxelCount() << std::endl;
+         //std::cerr << "boundary count = " << fs.boundaryCount() << std::endl;
+         fs.sweep();
+         auto grid2 = fs.sdfGrid();
+#ifdef TIMING_FAST_SWEEPING
+         timer.stop();
+#endif
+         writeFile("/tmp/bunny_mask_output.vdb", grid2);
+         {// Check that the norm of the gradient for all active voxels is close to unity
+             tools::Diagnose<FloatGrid> diagnose(*grid2);
+             tools::CheckNormGrad<FloatGrid> test(*grid2, 0.99f, 1.01f);
+             const std::string message = diagnose.check(test,
+                                                        false,// don't generate a mask grid
+                                                        true,// check active voxels
+                                                        false,// ignore active tiles since a level set has none
+                                                        false);// no need to check the background value
+             //std::cerr << message << std::endl;
+             const double percent = 100.0*double(diagnose.failureCount())/double(grid2->activeVoxelCount());
+             //std::cerr << "Failures = " << percent << "%" << std::endl;
+             //std::cerr << "Failed: " << diagnose.failureCount() << std::endl;
+             //std::cerr << "Total : " << grid2->activeVoxelCount() << std::endl;
+             //CPPUNIT_ASSERT(message.empty());
+             //CPPUNIT_ASSERT_EQUAL(size_t(0), diagnose.failureCount());
+             CPPUNIT_ASSERT(percent < 4.5);// crossing characteristics!
+             //std::cout << "\nOutput 1: " << message << std::endl;
+         }
+     }
+#endif
+}// testMaskSdf
+
+void
+TestFastSweeping::testFogToSdf()
+{
```

Comment:
This test never actually calls `fogToSdf`. should probably call it at least once to make sure it can be invoked

Reply:
this unit-test should be renamed testSdfToFog :) Additionally, I should add a testFogToSdf

Reply:
fixed this